tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AWS/SNS/PublishBatchAsyncIntegration.cs

```diff
@@ -0,0 +1,79 @@
+// <copyright file="PublishBatchAsyncIntegration.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.ComponentModel;
+using System.Threading;
+using Datadog.Trace.ClrProfiler.CallTarget;
+using Datadog.Trace.DuckTyping;
+using Datadog.Trace.Tagging;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.AWS.SNS
+{
+    /// <summary>
+    /// AWSSDK.SNS PublishBatchAsync CallTarget instrumentation
+    /// </summary>
+    [InstrumentMethod(
+        AssemblyName = "AWSSDK.SimpleNotificationService",
+        TypeName = "Amazon.SimpleNotificationService.AmazonSimpleNotificationServiceClient",
+        MethodName = "PublishBatchAsync",
+        ReturnTypeName = "System.Threading.Tasks.Task`1[Amazon.SimpleNotificationService.Model.PublishBatchResponse]",
+        ParameterTypeNames = new[] { "Amazon.SimpleNotificationService.Model.PublishBatchRequest", ClrNames.CancellationToken },
+        MinimumVersion = "3.0.0",
+        MaximumVersion = "3.*.*",
+        IntegrationName = AwsSnsCommon.IntegrationName)]
+    [Browsable(false)]
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public class PublishBatchAsyncIntegration
+    {
+        private const string Operation = "PublishBatch";
+
+        /// <summary>
+        /// OnMethodBegin callback
+        /// </summary>
+        /// <typeparam name="TTarget">Type of the target</typeparam>
+        /// <typeparam name="TPublishBatchRequest">Type of the request object</typeparam>
+        /// <param name="instance">Instance value, aka `this` of the instrumented method</param>
+        /// <param name="request">The request for the SNS operation</param>
+        /// <param name="cancellationToken">CancellationToken value</param>
+        /// <returns>CallTarget state value</returns>
+        internal static CallTargetState OnMethodBegin<TTarget, TPublishBatchRequest>(TTarget instance, TPublishBatchRequest request, CancellationToken cancellationToken)
+        {
+            if (request is null)
+            {
+                return CallTargetState.GetDefault();
+            }
+
+            var requestProxy = request.DuckCast<IPublishBatchRequest>();
+
+            var scope = AwsSnsCommon.CreateScope(Tracer.Instance, Operation, SpanKinds.Producer, out AwsSnsTags tags);
+            tags.TopicArn = requestProxy.TopicArn;
+            tags.TopicName = AwsSnsCommon.GetTopicName(requestProxy.TopicArn);
+
+            if (scope?.Span.Context != null)
+            {
+                ContextPropagation.InjectHeadersIntoBatch<TPublishBatchRequest>(requestProxy, scope.Span.Context);
```

Comment:
nit: if you add the where condition to `InjectHeadersIntoBatch` (previous comments) you can remove the type arg here

```suggestion
                ContextPropagation.InjectHeadersIntoBatch(requestProxy, scope.Span.Context);
```

Reply:
Resolving due to [this comment](https://github.com/DataDog/dd-trace-dotnet/pull/4712#issuecomment-1762212524).